api/src/org/labkey/api/pipeline/PathSerialization.java

```
@@ -71,14 +74,24 @@ protected Deserializer()
         public Path deserialize(JsonParser parser, DeserializationContext context) throws IOException
         {
             String str = parser.getValueAsString();
+//            return Path.of(URI.create(str));
             if (FileUtil.hasCloudScheme(str))
             {
-                // TODO: problem is that we need a container to map a URL string to an S3 path, because we have a prefix derived form the container (#35865)
-                // TODO: one possibility is to tease out what config/container matches the bucket/prefix we find, but there could be more than 1 match
-                return null;
+//                // TODO: problem is that we need a container to map a URL string to an S3 path, because we have a prefix derived form the container (#35865)
+//                // TODO: one possibility is to tease out what config/container matches the bucket/prefix we find, but there could be more than 1 match
+//                return Path.of(str);
+                CloudStoreService css = CloudStoreService.get();
+                if (css != null)
+                {
+                    //TODO this will likely work only for cloud paths that include access ids eg: s3://<pub-access-key>@s3.amazonaws.com/<my bucket>/...
+                    return css.getPathFromUrl(str);
+                    //TODO need LKS container to pull the appropriate S3 credentials/FileSystem representation (in the event of interrupted job)
+                }
+
+                return Path.of(str); //TODO this will cause issues...
```

would it be better to throw then?

---

If the string path isn't viable it will cause an exception, the existing message is fairly clear (at least as clear as I would make it) rather than catch and rethrow, I think it makes sense to leave it as is and let the parser throw if needed.